src/network/networkd-manager-varlink.c

```diff
@@ -259,6 +264,66 @@ static int vl_method_set_persistent_storage(sd_varlink *vlink, sd_json_variant *
         return sd_varlink_reply(vlink, NULL);
 }
 
+static int vl_method_set_link_down(sd_varlink *vlink, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        Manager *manager = ASSERT_PTR(userdata);
+        Link *link = NULL;
+        int r;
+
+        assert(vlink);
+
+        r = dispatch_interface(vlink, parameters, manager, &link);
+        if (r != 0)
+                return r;
+
+        r = varlink_verify_polkit_async(
+                        vlink,
+                        manager->bus,
+                        "org.freedesktop.network1.manage-links",
+                        /* details= */ NULL,
+                        &manager->polkit_registry);
+        if (r <= 0)
+                return r;
+
+        /* Stop all network engines while interface is still up, then bring it down */
+        r = link_stop_engines(link, /* may_keep_dynamic = */ false);
+        if (r < 0)
+                log_link_warning_errno(link, r, "Failed to stop network engines: %m");
+
+        /* Now bring the interface down via netlink */
+        r = link_request_to_bring_up_or_down(link, false);
```

Comment:
Well,,, I do not think it is not necessary to queue the request, but call RTM_SETLINK now.
More importantly, the varlink server should reply the result after the RTM_SETLINK call finishes.
So, please introduce, say `link_up_or_down_now_by_varlink()` or so, which is similar to `link_up_or_down_now()` but takes a simple struct that contains `Link *` and `sd_varlink *` as a userdata for `netlink_call_async()`.